### DISCUSSION TITLE
visualizing spans in a notebook displacy.serve vs displacy.render

### DISCUSSION THREAD

**Discussion by joseberlines** [explosion/spaCy#11305] [Help: Other Questions] [answered]
I am following the instructions of spacy in order to visualize the spans in a notebook and copying the code : https://spacy.io/usage/visualizers#span

```
import spacy
from spacy import displacy
from spacy.tokens import Span

text = "Welcome to the Bank of China."

nlp = spacy.blank("en")
doc = nlp(text)

doc.spans["sc"] = [
    Span(doc, 3, 6, "ORG"), 
    Span(doc, 5, 6, "GPE"),
]

displacy.serve(doc, style="span")
```
the result in my notebook is as follows:
![image](https://user-images.githubusercontent.com/33203439/184589374-da2784bf-15d3-4e4a-8b64-e2a75f9f1ddb.png)

it is weird that when using displacy.render the rendering is of another quality as when using "serve". Serve can not be used properly in a notebook.

I would like to know what I need to do to get the result shown in the documentation within a notebook. That means, the underlying.

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @joseberlines , may I know what version your spaCy is? I can't seem to replicate it in both `jupyter notebook` and `jupyter lab` environments with spaCy v3.4. We encountered the same error before (c.f. https://github.com/explosion/spaCy/issues/11040) and was able to have a fix (c.f. https://github.com/explosion/spaCy/pull/10994) in v3.4. I highly encourage upgrading and trying again. You can check your spaCy version via:

```
python -m spacy info
```

In any case, usually you'd find a scrollbar on the rightmost part of the output cell that "hides" these span labels. The fix in spaCy v3.4 removes the need to scroll though.

**Reply by joseberlines** [reply]
spaCy version    3.3.0

**Reply by joseberlines** [reply]
you are right about the scroll bar, I did not even realised, still I get something like:
![image](https://user-images.githubusercontent.com/33203439/184607553-3ab9b7e4-44d6-4f23-baea-2377ff67ea41.png)

**Reply by joseberlines** [reply]
I dont know if the problem is on my side or something could be improve in the spacy side

**Reply by joseberlines** [reply]
we also use voila with jupyter lab and the result when rendering with voila is:
![image](https://user-images.githubusercontent.com/33203439/184607932-a20aabe1-f267-4f0c-be78-4c147008b69b.png)

**Reply by ljvmiranda921** [reply]
Yes, I think v3.4.0 also solves the same problem. Can you try upgrading and see if this issue still persists? (Unless these screenshots are already in v3.4.0)

**Reply by joseberlines** [reply]
Thanks. I will update and report.
